zwave_js_server/model/node.py

```diff
@@ -322,35 +352,31 @@ async def async_set_value(
             raise UnwriteableValue
 
         # the value object needs to be send to the server
-        args = {
-            "command": "node.set_value",
-            "nodeId": self.node_id,
-            "valueId": val.data,
-            "value": new_value,
-        }
-        if wait_for_result:
-            result = await self.client.async_send_command(args)
-            return cast(bool, result)
-        await self.client.async_send_command_no_wait(args)
-        return None
+        result = await self._async_send_command(
+            "set_value",
+            valueId=val.data,
+            value=new_value,
+            wait_for_result=wait_for_result,
+        )
+
+        if result is None:
+            return None
+
+        return isinstance(result, dict) and result.get("success", False)
```

Comment:
We know the result type for this command. I think we should blow up if that would be incorrect.

https://github.com/zwave-js/zwave-js-server/blob/c1ddbd1c802712161d8cab2020f1a5a1058d6453/src/lib/node/outgoing_message.ts#L5

```suggestion
        return result["success"]
```